src/components/UsedCoupons/index.jsx

```diff
@@ -0,0 +1,216 @@
+import React, { useState, useEffect, useRef } from "react";
+import "./style.css";
+import {
+  Table,
+  Input,
+  Button,
+  Space,
+  Typography,
+  Spin,
+  Divider,
+  message,
+} from "antd";
+import Highlighter from "react-highlight-words";
+import { SearchOutlined } from "@ant-design/icons";
+import { useTranslation } from "react-i18next";
+import firebase from "../../firebaseConfig";
+
+const db = firebase.firestore();
+const { Title } = Typography;
+
+const UsedCoupons = ({ isTesting }) => {
+  const { t } = useTranslation();
+  const [usedCoupons, setUsedCoupons] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [searchState, setSearchState] = useState({
+    searchText: "",
+    searchedColumn: "",
+  });
+  const docID = useRef(null);
+  let searchInput;
+
+  const getDocId = async () => {
+    // get the id of the doc to fetch later in snapshot
+    const res = await db
+      .collection("sampleSoldCoupons")
+      .where("compName", "==", "Arby's")
+      .get();
+    const data = res.docs.map((x) => x.id);
+    [docID.current] = data;
+  };
+
+  useEffect(() => {
+    getDocId().then(() => {
+      const unsubscribe = db
+        .collection("sampleSoldCoupons") // this collection is a test, we don't have the real collection yet
+        .doc(docID.current)
+        .collection("coupons")
+        .where("isActive", "==", false)
+        .onSnapshot((snapshot) => {
+          const dataArr = [];
+          snapshot.forEach((doc) => {
+            dataArr.push({
+              ...doc.data(),
+              key: doc.id,
+              // convert the timestamp into a usable date
+              date:
+                doc.data().markDate &&
+                `${doc.data().markDate.toDate().getDate()}/${
+                  doc.data().markDate.toDate().getMonth() + 1
+                }/${doc.data().markDate.toDate().getFullYear()}`,
+            });
+          });
+          setUsedCoupons(dataArr);
+          setLoading(false);
+        });
+      return unsubscribe;
+    });
+  }, []);
+
+  const handleSearch = (selectedKeys, confirm, dataIndex) => {
+    confirm();
+    setSearchState({
+      searchText: selectedKeys[0],
+      searchedColumn: dataIndex,
+    });
+  };
+
+  const handleReset = (clearFilters) => {
+    clearFilters();
+    setSearchState({ searchText: "" });
+  };
+
+  const getColumnSearchProps = (dataIndex) => ({
```

Comment:
can we move this to another file and import it from there

Reply:
You mean the getColumnSearchProps function? @halitbatur